ui/src/app/calendar/calendar.service.ts

```diff
@@ -193,11 +201,13 @@ export class CalendarService {
         const weekdays = ['SUNDAY', 'MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY'];
         const regularDays = room.defaultWorkingHours.map((d) => weekdays.indexOf(d.weekday));
         const extraDays = room.calendarExceptionEvents
-            .filter((e) => !e.outOfService && e.startDate >= start && e.endDate <= end)
+            .filter((e) => !e.outOfService && this.weeksOverlap(e.startDate, e.endDate, start, end))
```

Comment:
Miks tässä tarkistetaan nyt meneekö viikot päällekkäin? Kun ennen testattiin meneekö aikaikkunat päällekkäin.

Reply:
Tuossa weeksOverlap-funktiossa. Katsotaan mätsääkö poikkeusaukiolo kyseiseen viikkoon. Aikaisemmassa oli bugi jos poikkeus jatkuu seuraavalle viikolle niin sitä ei huomioitu.